app/lib/staypuft/network_query.rb

```diff
@@ -155,6 +155,7 @@ def interface_hash_for_host(subnet_type_name, host)
     def gateway_hash_for_host(host)
       gateway_hash = interface_hash_for_host(Staypuft::SubnetType::PUBLIC_API, host)
       gateway_hash = interface_hash_for_host(Staypuft::SubnetType::PXE, host) unless gateway_hash[:subnet]
+      gateway_hash = interface_hash_for_host(Staypuft::SubnetType::EXTERNAL, host) if Staypuft::Role.compute.any? { |r| r.name == host.hostgroup.name } && host.deployment.nova_networking?
```

Comment:
When the deployment network type is Nova, the external interface is a requirement on the compute node(s). We don't want to set it to PXE.